code/modules/mob/living/simple_animal/bot_swarm/swarmer.dm

```diff
@@ -0,0 +1,601 @@
+			var/obj/machinery/computer/C = target
+			src << "[C][C.circuit]"
+			if(C.circuit)
+				C.circuit.loc = M.loc
+		qdel(target)
+
+
+/obj/effect/swarmer //Default swarmer effect object visual feedback
+	name = "swarmer ui"
+	desc = null
+	gender = NEUTER
+	icon = 'icons/mob/swarmer.dmi'
+	icon_state = "ui_light"
+	mouse_opacity = 0
+	layer = 4
+	unacidable = 1
+
+/obj/effect/swarmer/disintegration
+	icon_state = "disintegrate"
+
+/obj/effect/swarmer/disintegration/New()
+	playsound(src.loc, "sparks", 100, 1)
+	spawn(10)
+		qdel(src)
+
+/obj/effect/swarmer/dismantle
+	icon_state = "dismantle"
+
+/obj/effect/swarmer/dismantle/New()
+	spawn(25)
+		qdel(src)
+
+/obj/effect/swarmer/integrate
+	icon_state = "integrate"
+
+/obj/effect/swarmer/integrate/New()
+	spawn(5)
+		qdel(src)
+
+/obj/effect/swarmer/destructible //Default destroyable object for swarmer constructions
+	luminosity = 1
+	mouse_opacity = 1
+	var/health = 30
+
+/obj/effect/swarmer/destructible/proc/TakeDamage(damage)
+	health -= damage
+	if(health <= 0)
+		qdel(src)
+
+/obj/effect/swarmer/destructible/bullet_act(obj/item/projectile/Proj)
+	if(Proj.damage)
+		if((Proj.damage_type == BRUTE || Proj.damage_type == BURN))
+			TakeDamage(Proj.damage)
+	..()
+
+/obj/effect/swarmer/destructible/attackby(obj/item/weapon/I, mob/living/user, params)
+	if(istype(I, /obj/item/weapon))
+		user.changeNext_move(CLICK_CD_MELEE)
+		user.do_attack_animation(src)
+		TakeDamage(I.force)
+	return
+
+/obj/effect/swarmer/destructible/ex_act()
+	qdel(src)
+	return
+
+/obj/effect/swarmer/destructible/blob_act()
+	qdel(src)
+	return
+
+/obj/effect/swarmer/destructible/emp_act()
+	qdel(src)
+	return
+
+/obj/effect/swarmer/destructible/attack_animal(mob/living/user)
+	if(isanimal(user))
+		var/mob/living/simple_animal/S = user
+		S.do_attack_animation(src)
+		user.changeNext_move(CLICK_CD_MELEE)
+		if(S.melee_damage_type == BRUTE || S.melee_damage_type == BURN)
+			TakeDamage(rand(S.melee_damage_lower, S.melee_damage_upper))
+	return
+
+/mob/living/simple_animal/hostile/swarmer/proc/CreateTrap()
+	set name = "Create trap"
+	set category = "Swarmer"
+	set desc = "Creates a simple trap that will non-lethally electrocute anything that steps on it. Costs 5 resources"
+	if(/obj/effect/swarmer/destructible/trap in loc)
+		src << "<span class='warning'>There is already a trap here. Aborting.</span>"
+		return
+	Fabricate(/obj/effect/swarmer/destructible/trap, 5)
+	return
+
+/obj/effect/swarmer/destructible/trap
+	name = "swarmer trap"
+	desc = "A quickly assembled trap that electrifies living beings and overwhelms machine sensors. Will not retain its form if damaged enough."
+	icon_state = "trap"
+	luminosity = 1
+	health = 10
+
+/obj/effect/swarmer/destructible/trap/Crossed(var/atom/movable/AM)
+	if(isliving(AM))
+		var/mob/living/L = AM
+		if(!istype(L, /mob/living/simple_animal/hostile/swarmer))
+			playsound(loc,'sound/effects/snap.ogg',50, 1, -1)
+			L.electrocute_act(0, src, 1, def_zone = "l_foot",1)
```

Comment:
Change this to: L.electrocute_act(0, src, 1, "l_foot", 1). No need to define the argument, and argument definitions in proc calls need to be at the end of the proc call.